### ISSUE TITLE
policy-agent lose connection to kube-apiserver forever

### ISSUE THREAD

**Issue by aborilov** [projectcalico/k8s-policy#54]
If policy-agent loose connection to kube-apiserver(for example, on kube-apiserver restart), it never gets back connection. After policy-agent restart, all works fine again.
```
2016-11-02 07:22:36,156 5 INFO Emptied the event queue
2016-11-02 07:22:48,928 5 WARNING Re-starting watch on resource: Pod
2016-11-02 07:22:48,929 5 WARNING Re-starting watch on resource: NetworkPolicy
2016-11-02 07:22:48,931 5 WARNING Re-starting watch on resource: Namespace
Exception in thread Thread-3:
Traceback (most recent call last):
  File "threading.py", line 801, in __bootstrap_inner
  File "threading.py", line 754, in run
  File "<string>", line 286, in _manage_resource
  File "<string>", line 320, in _watch_resource
  File "site-packages/requests/models.py", line 702, in iter_lines
  File "site-packages/requests/models.py", line 663, in generate
ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', IncompleteRead(0 bytes read))
```

**Comment by heschlie** [comment]
@aborilov What version of the policy-controller are you using?  I believe this exception was fixed already, when running the current build I am able to restart, and even stop, the kube-apiserver.

When I run the calico/kube-policy-controller:v0.3.0 or v0.4.0 I see it recover, as we started catching that exception in PR #27. This PR however also changed from using the v1alpha1 k8s API endpoint to v1beta1, so if you are using an older version of k8s you might not be able to upgrade to the latest policy-controller without first upgrade k8s.
